OutOfSchool/OutOfSchool.WebApi/Services/Database/WorkshopService.cs

```diff
@@ -266,22 +269,100 @@ public async Task<SearchResult<T>> GetByProviderId<T>(Guid id, ExcludeIdFilter f
             filter.ExcludedId == null
                 ? (x.ProviderId == id)
                 : (x.ProviderId == id && x.Id != filter.ExcludedId)).ConfigureAwait(false);
-        var workshops = await workshopRepository.Get(
+
+        var workshops = workshopRepository.Get(
                 skip: filter.From,
                 take: filter.Size,
                 includeProperties: includingPropertiesForMappingDtoModel,
                 where: x => filter.ExcludedId == null
                     ? (x.ProviderId == id)
-                    : (x.ProviderId == id && x.Id != filter.ExcludedId))
-            .ToListAsync()
-            .ConfigureAwait(false);
-
-        logger.LogInformation(!workshops.Any()
+                    : (x.ProviderId == id && x.Id != filter.ExcludedId));
+
+        var chatrooms = roomRepository.Get(
+            includeProperties: $"{nameof(ChatRoomWorkshop.ChatMessages)}",
+            where: x => x.ChatMessages.Where(x => x.ReadDateTime == null && !x.SenderRoleIsProvider).Any());
```

Comment:
GitHub Action says `The best overload for 'Get' does not have a parameter named 'where'`. Probably it should be `whereExpression`?